### PR TITLE
Dist-DL: Adding user guides for deepspeed, fsdp and ray-tune

### PR DESCRIPTION
**Pull request by AhmedYSoliman**
## Objective
This PR is to introduce 3 user guides for Distributed DL topics: 
1. Deep Learning Optimization with Microsoft DeepSpeed
2. Fine-Tuning with Fully-Sharded Data Parallel (FSDP)
3. Automated Hyperparameter Optimization (HPO) with Ray Tune

Based on the experiments tutorial in the following repos:
- [Kaust-rccl/Dist-DL-training](https://github.com/kaust-rccl/Dist-DL-training)
- [Kaust-rccl/HPO-With-Ray](https://github.com/kaust-rccl/hpo-with-ray)

## Modification
1. Added 3 rst files for each user guide under `science platform/data science platform/Distributed ML/DL on KSL systems`
2. Old pages for DeepSpeed and Ray-tune are keep; They contain some demo experiments and Shaheen related experiment.
3. Below is the structure of the index.rst file for `Distributed ML/DL on KSL`

image

## Related Tickets
- [RT: 63620](https://tickets.hpc.kaust.edu.sa/Ticket/Display.html?id=63620)

## Important Points Before Accepting the Merge PR
1. Pending PR merge approval in [Dist-DL-training - PR#15](https://github.com/kaust-rccl/Dist-DL-training/pull/15)
3. Pending PR merge approval in [HPO-with-ray - PR#2](https://github.com/kaust-rccl/hpo-with-ray/pull/2)
5. Unresolved [issue #16](https://github.com/kaust-rccl/Dist-DL-training/issues/16) in Dist-DL-training repo, for Nan values in the evaluation loss results during running experiments for multi-gpus and multi nodes.

## Points to Mention
1.  In the user guides, I've added reference to the associated repos for more deep details and illustration, so the user should follow.
2. Output results were not added for FSDP due to the unresolved issue mentioned above.
3. Some hyperlinks in hpo ray tune user guide will not open till its merge PR is approved, because they are refering to the main branch in the repo.

### CODE REVIEW

**Review by D-Barradas — APPROVED** [review]
The fsdp gide have some details that can be fixed later after merging , for example 
in line 167 the name of the file is wrong , is single_node_<jobid>.out 
The conda path is a good exerces for the multiGPU, I have pushed some changes in the Repo to refelec this exercise 
Also I fixed the conda enviroment fro Bloom
